### PR TITLE
TRUS-4357 Add hidden caption to headingWithCaption

### PR DESCRIPTION
**Pull request by carrielies**

### CODE REVIEW

File: test/views/behaviours/ViewBehaviours.scala [hunk 1]
```diff
@@ -88,10 +88,10 @@ trait ViewBehaviours extends ViewSpecBase {
           assertEqualsMessage(doc, "title", s"$messageKeyPrefix.title")
         }
 
-        "display the correct page title with caption" in {
+        "display the correct page title with section" in {
 
           val doc = asDocument(view)
-          assertPageTitleWithCaptionEqualsMessages(doc, s"$messageKeyPrefix.caption",  captionParam, s"$messageKeyPrefix.heading")
+          assertPageTitleWithSectionSubheading(doc, s"$messageKeyPrefix", captionParam = "")
```
Comment (miquelht) [inline test/views/behaviours/ViewBehaviours.scala:94]: messageKeyPrefix is just a String so

s"$messageKeyPrefix" -> messageKeyPrefix

File: app/views/ViewUtils.scala [hunk 2]
```diff
@@ -58,7 +58,7 @@ object ViewUtils {
           value = Some(a.value),
           checked = field.value.contains(a.value),
           content = Text(messages(a.messageKey)),
-          attributes = if (trackGa) Map[String, String]("data-journey-click" -> s"trusts-frontend:click:${a.id}") else Map.empty
```
Comment (miquelht) [inline app/views/ViewUtils.scala:61]: As the only usage of trackGa has been removed the parameter can be removed from the method signature and usages

File: test/views/ViewSpecBase.scala [hunk 3]
```diff
@@ -66,6 +66,28 @@ trait ViewSpecBase extends SpecBase {
 
   }
 
+  def assertPageTitleWithSectionSubheading(doc: Document,
+                                           expectedMessageKey: String,
+                                           captionParam: String,
+                                           args: Any*): Assertion = {
```
Comment (miquelht) [inline test/views/ViewSpecBase.scala:72]: By looking at the usage of this method captionParam and args are both redundant and can be removed
This
${messages(s"$expectedMessageKey.caption", captionParam) and just be
${messages(s"$expectedMessageKey.caption")

Similarly
     val expectedHeading = messages(s"$expectedMessageKey.heading", args:_*)
becomes
     val expectedHeading = messages(s"$expectedMessageKey.heading")

File: test/views/behaviours/ViewBehaviours.scala (hunk 1 above)
Comment (carrielies) [reply test/views/behaviours/ViewBehaviours.scala]: Done

File: test/views/ViewSpecBase.scala (hunk 3 above)
Comment (carrielies) [reply test/views/ViewSpecBase.scala:72]: I didn't change this one as it's in the test and I'd already applied the other changes to all the projects

File: app/views/ViewUtils.scala (hunk 2 above)
Comment (carrielies) [reply app/views/ViewUtils.scala:61]: Done

**Review by miquelht — APPROVED** [review]

**Pull request merged**